app/models/concerns/due_date_queries.rb

```diff
@@ -1,291 +1,45 @@
-  # Get upcoming deadlines with limit
-  def upcoming_deadlines(limit: 5)
-    due_dates.upcoming.limit(limit)
-  end
-
-  # Get overdue deadlines
-  def overdue_deadlines
-    due_dates.overdue
-  end
-
-  # Check if there are any future deadlines
-  def has_future_deadlines?
-    due_dates.upcoming.exists?
-  end
-
-  # Get deadlines for a specific round
-  def deadlines_for_round(round_number)
-    due_dates.where(round: round_number).order(:due_at)
-  end
-
-  # Find deadline by type and round
-  def find_deadline(deadline_type_name, round_number = nil)
-    query = due_dates.joins(:deadline_type).where(deadline_types: { name: deadline_type_name })
-    query = query.where(round: round_number) if round_number
-    query.order(:due_at).first
-  end
-
-  # Get all deadline types used by this object
-  def used_deadline_types
-    due_dates
-      .joins(:deadline_type)
-      .select('DISTINCT deadline_types.*')
-      .map(&:deadline_type)
-  end
-
-  # Check if this object has a specific type of deadline
-  def has_deadline_type?(deadline_type_name)
-    due_dates
-      .joins(:deadline_type)
-      .where(deadline_types: { name: deadline_type_name })
-      .exists?
-  end
-
-  # Get deadlines that are currently active (allowing some action)
-  def active_deadlines
-    due_dates.select(&:active?)
-  end
-
-  # Get deadline summary for display
-  def deadline_summary
-    {
-      total_deadlines: due_dates.count,
-      upcoming_count: upcoming_deadlines.count,
-      overdue_count: overdue_deadlines.count,
-      deadline_types: used_deadline_types.map(&:name),
-      next_deadline: next_due_date,
-      has_active_deadlines: active_deadlines.any?
-    }
-  end
-
-  # Find the current stage for topic-specific deadlines
-  def current_stage_for_topic(topic_id, action)
-    deadline_type_name = map_action_to_deadline_type(action)
-    return nil unless deadline_type_name
-
-    # Try topic-specific deadline first
-    topic_deadline = due_dates
-                     .joins(:deadline_type)
-                     .where(parent_id: topic_id, parent_type: 'SignUpTopic')
-                     .where(deadline_types: { name: deadline_type_name })
-                     .where('due_at >= ?', Time.current)
-                     .order(:due_at)
-                     .first
+  def next_due_date(topic_id = nil)
+    if topic_id && has_topic_specific_deadlines?
+      topic_deadline = due_dates.where(parent_id: topic_id, parent_type: 'SignUpTopic')
+                               .where('due_at >= ?', Time.current)
+                               .order(:due_at)
+                               .first
+      return topic_deadline if topic_deadline
+    end
 
-    # Fall back to assignment-level deadline
-    topic_deadline || current_deadline_for(action)
+    due_dates.where('due_at >= ?', Time.current).order(:due_at).first
   end
 
-  # Get all deadlines affecting a specific topic
+  # Get all deadlines for a topic (topic-specific + assignment fallback)
```

Comment:
I don't see any need to collect all deadlines for a topic.  Another case of YAGNI.